dotcom-rendering/src/components/ScrollingDots.tsx

```diff
@@ -0,0 +1,169 @@
+import { css } from '@emotion/react';
+import { palette } from '../palette';
+
+const dotSize = 7;
+const dotGap = 4;
+const dotsVisible = 5;
+const scrollThreshold = Math.floor(dotsVisible / 2);
+
+const scrollingDotContainerStyles = css`
+	overflow: hidden;
+	max-width: ${dotSize * dotsVisible + dotGap * (dotsVisible - 1) + 8}px;
+`;
+
+const scrollingDotStyles = (total: number) => css`
+	display: grid;
+	gap: ${dotGap}px;
+	grid-template-columns: repeat(${total}, ${dotSize}px);
+	padding: 4px;
+	transition: transform 0.25s ease;
+`;
+
+const dotStyles = css`
+	width: ${dotSize}px;
+	height: ${dotSize}px;
+	border-radius: 100%;
+	background-color: ${palette('--slideshow-pagination-dot')};
+	transition: all 0.25s ease;
+`;
+
+const activeDotStyles = css`
+	transform: scale(1.15);
+	background-color: ${palette('--slideshow-pagination-dot-active')};
+`;
+
+const trailingDotStyles = css`
+	transform: scale(0.85);
+`;
+
+const offscreenDotStyles = css`
+	transform: scale(0);
+`;
+
+export const ScrollingDots = ({
```

Comment:
I guess this should only be used in the slideshow carousels... wonder if it would be worth indicating that in the component name? Something like, maybe: 

```suggestion
export const SlideshowCarouselScrollingDots = ({
```

Reply:
It's generic enough that it _should_ be useable elsewhere. That said, perhaps it is best to keep the name tied to slideshows for now to keep things grouped together.

Reply:
Yeah, agree it's generic enough so certainly could be used elsewhere. Wasn't sure of what those use cases would be though! 

Not too fussed about it, so up to you!

Reply:
Can't think of any use cases outside of carousels, and I think we've added enough of those for now 😅 If it's a standalone reusable component then it should probably have its own stories etc. too. I've renamed it as per your suggestion to indicate it's a constituent component of the slideshow carousels. We can always rename it in the future if it turns out to be useful elsewhere.